google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/apis/GoogleCloudApiSelectorPanel.java

```diff
@@ -147,7 +181,7 @@ private void createUIComponents() {
                 int selectedIndex = model.getMinSelectionIndex();
                 CloudLibrary library =
                     (CloudLibrary) cloudLibrariesTable.getModel().getValueAt(selectedIndex, 0);
-                detailsPanel.setCloudLibrary(library);
+                detailsPanel.setCloudLibrary(library, apiManagementMap.get(library));
```

Comment:
So this essentially sets a library and its "enable on cloud" state to details panel, and `apiManagementMap` is always pre-populated with the set of libraries and they match, so no chance to get null here, correct? This maybe a candidate for immutable map?

Reply:
correct. I made it immutable. thanks